components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,632 @@
+            long currentTimeInMillis = System.currentTimeMillis();
+            long iatValidityPeriodInMillis = iatValidityPeriod * 1000;
+
+            if (currentTimeInMillis - issuedAtTimeMillis > iatValidityPeriodInMillis) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Logout token is used after iat validity period." +
+                            ", iat validity period(m) : " + iatValidityPeriod +
+                            ", Current Time : " + currentTimeInMillis +
+                            ". Token Rejected and validation terminated.");
+                }
+                throw handleLogoutClientException(ErrorMessages.LOGOUT_TOKEN_IAT_VALIDATION_FAILED);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("iat validity period of Token was validated successfully.");
+            }
+        }
+    }
+
+    /**
+     * Get iatValidityPeriod from configuration file.
+     * Use default value if error occurs.
+     *
+     * @return - iatValidityPeriod.
+     */
+    private long getIatValidityPeriod() {
+
+        long iatValidityPeriod = 0;
+        if (StringUtils.isBlank(getAuthenticatorConfig().getParameterMap()
+                .get(OIDCAuthenticatorConstants.BackchannelLogout.IAT_VALIDITY_PERIOD))) {
+            iatValidityPeriod = DEFAULT_IAT_VALIDITY_PERIOD;
+        } else {
+            try {
+                iatValidityPeriod =
+                        Integer.parseInt(getAuthenticatorConfig().getParameterMap()
+                                .get(OIDCAuthenticatorConstants.BackchannelLogout.IAT_VALIDITY_PERIOD));
+            } catch (NumberFormatException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Invalid iatValidityPeriod in config file. Using default iatValidityPeriod value");
+                }
+            }
+        }
+        return iatValidityPeriod;
+    }
+
+    /**
+     * Do the sid claim validation.
+     *
+     * @param claimsSet
+     * @return boolean if validation is successful.
+     */
+    private boolean isSidClaimExists(JWTClaimsSet claimsSet) {
+
+        return StringUtils.isNotBlank((String) claimsSet.getClaim(OIDCAuthenticatorConstants.Claim.SID));
+    }
+
+    /**
+     * Do the sub claim validation.
+     *
+     * @param claimsSet
+     * @return boolean if validation is successful.
+     */
+    private boolean isSubClaimExists(JWTClaimsSet claimsSet) {
+
+        return StringUtils.isNotBlank(claimsSet.getSubject());
+    }
+
+    /**
+     * Do the event claim validation according to OIDC back-channel logout specification.
+     *
+     * @param event
+     * @return boolean if validation is successful.
+     * @throws LogoutClientException
+     */
+    private void validateEventClaim(JSONObject event) throws LogoutClientException {
+
+        if (event == null ||
+                !StringUtils.equals(event.getAsString(OIDCAuthenticatorConstants.Claim.BACKCHANNEL_LOGOUT_EVENT),
```

Comment:
Get the event claim to a variable and do the comparison. It will be much readable

WDYT?